### PR TITLE
feat: apply mrm state to awapi

### PR DESCRIPTION
**Pull request by isamu-takagi**
Signed-off-by: Takagi, Isamu <isamu.takagi@tier4.jp>

## PR Type

- Improvement

## Related Links

https://github.com/autowarefoundation/autoware.universe/pull/2070

## Description

Apply MRM state.

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [ ] Code follows [coding guidelines][coding-guidelines]
- [ ] Assign PR to reviewer

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] Code follows [coding guidelines][coding-guidelines]
- [ ] (Optional) Unit tests have been written for new behavior
- [ ] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [ ] All open points are addressed and tracked via issues or tickets
- [ ] Write [release notes][release-notes]

## CI Checks

- **Build and test for PR / build-and-test-pr**: Required to pass before the merge.
- **Build and test for PR / clang-tidy-pr**: NOT required to pass before the merge. It is up to the reviewer(s).
- **Check spelling**: NOT required to pass before the merge. It is up to the reviewer(s). See [here][spell-check-dict] if you want to add some words to the spell check dictionary.

[coding-guidelines]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/1194394777/T4
[release-notes]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/563774416
[spell-check-dict]: https://github.com/tier4/autoware-spell-check-dict#how-to-contribute

### CODE REVIEW

File: awapi_awiv_adapter/src/awapi_awiv_adapter_core.cpp [hunk 1]
```diff
@@ -84,7 +84,7 @@ AutowareIvAdapter::AutowareIvAdapter()
     "input/control_mode", 1, std::bind(&AutowareIvAdapter::callbackControlMode, this, _1));
   sub_gate_mode_ = this->create_subscription<tier4_control_msgs::msg::GateMode>(
     "input/gate_mode", durable_qos, std::bind(&AutowareIvAdapter::callbackGateMode, this, _1));
-  sub_emergency_ = this->create_subscription<autoware_auto_system_msgs::msg::EmergencyState>(
+  sub_emergency_ = this->create_subscription<autoware_adapi_v1_msgs::msg::MrmState>(
     "input/emergency_state", 1, std::bind(&AutowareIvAdapter::callbackEmergencyState, this, _1));
```
Comment (mkuri) [inline awapi_awiv_adapter/src/awapi_awiv_adapter_core.cpp:88]: would you change callbackEmergencyState to callbackMrmState?

File: awapi_awiv_adapter/include/awapi_awiv_adapter/awapi_awiv_adapter_core.hpp [hunk 2]
```diff
@@ -124,8 +124,7 @@ class AutowareIvAdapter : public rclcpp::Node
   void callbackControlMode(
     const autoware_auto_vehicle_msgs::msg::ControlModeReport::ConstSharedPtr msg_ptr);
   void callbackGateMode(const tier4_control_msgs::msg::GateMode::ConstSharedPtr msg_ptr);
-  void callbackEmergencyState(
-    const autoware_auto_system_msgs::msg::EmergencyState::ConstSharedPtr msg_ptr);
+  void callbackEmergencyState(const autoware_adapi_v1_msgs::msg::MrmState::ConstSharedPtr msg_ptr);
```
Comment (mkuri) [inline awapi_awiv_adapter/include/awapi_awiv_adapter/awapi_awiv_adapter_core.hpp]: Would you change callbackEmergencyState to callbackMrmState?

File: awapi_awiv_adapter/src/awapi_awiv_adapter_core.cpp [hunk 3]
```diff
@@ -250,7 +250,7 @@ void AutowareIvAdapter::callbackGateMode(
 }
 
 void AutowareIvAdapter::callbackEmergencyState(
```
Comment (mkuri) [inline awapi_awiv_adapter/src/awapi_awiv_adapter_core.cpp]: Would you change callbackEmergencyState to callbackMrmState?

File: awapi_awiv_adapter/include/awapi_awiv_adapter/awapi_awiv_adapter_core.hpp (hunk 2 above)
Comment (isamu-takagi) [reply awapi_awiv_adapter/include/awapi_awiv_adapter/awapi_awiv_adapter_core.hpp]: Done https://github.com/tier4/tier4_ad_api_adaptor/pull/82/commits/f34536c8f1013d08812444cd677db486dc8391e3.

File: awapi_awiv_adapter/src/awapi_awiv_adapter_core.cpp (hunk 3 above)
Comment (isamu-takagi) [reply awapi_awiv_adapter/src/awapi_awiv_adapter_core.cpp]: Done https://github.com/tier4/tier4_ad_api_adaptor/pull/82/commits/f34536c8f1013d08812444cd677db486dc8391e3.

**Review by mkuri — APPROVED** [review]
LGTM

**Pull request merged**